rust/automerge/src/op_tree.rs

```diff
@@ -292,7 +307,53 @@ impl OpTreeInternal {
         }
     }
 
-    pub(crate) fn seek_opid<'a>(
+    pub(crate) fn seek_map_op<'a>(
+        &'a self,
+        op: Op<'a>,
+        clock: Option<&Clock>,
+        osd: &'a OpSetData,
+    ) -> Option<FoundOpId<'a>> {
+        let pos = self.binary_search_by(osd, |o| o.key_cmp(op.key()).then_with(|| o.cmp(&op)));
+        let mut iter = self.iter();
+        let op2 = iter.nth(pos).map(|idx| idx.as_op(osd))?;
+        assert_eq!(op, op2);
+        let index = 0;
+        for e in iter.map(|idx| idx.as_op(osd)) {
+            if e.elemid_or_key() != op.elemid_or_key() {
+                break;
+            }
+
+            if e.visible_at(clock) {
+                return Some(FoundOpId {
+                    op,
+                    index,
+                    visible: false,
+                });
+            }
+        }
+        Some(FoundOpId {
+            op,
+            index,
+            visible: op.visible_at(clock),
+        })
+    }
+
+    pub(crate) fn seek_idx<'a>(
+        &'a self,
+        idx: OpIdx,
+        encoding: ListEncoding,
+        clock: Option<&Clock>,
+        osd: &'a OpSetData,
+    ) -> Option<FoundOpId<'a>> {
+        let op = idx.as_op(osd);
+        if op.key().is_map_key() {
```

Comment:
What's the reason to use `is_map_key` here instead of just matching on `key()`? The latter seems clearer to me.